src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -132,6 +136,11 @@ public function __construct(string $name, string $value = null, $expire = 0, ?st
         }
 
         $this->sameSite = $sameSite;
+
+        if (empty(static::$reservedChars) && empty(static::$reservedCharsEncoded)) {
```

Comment:
let's just do: `if (!self::$reservedChars) {`

Reply:
This is an array, why it's better to treat it like boolean?

Reply:
`if` is not a function, it's a language construct. This argument doesn't make sense.